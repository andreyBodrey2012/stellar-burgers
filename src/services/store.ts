import { configureStore } from '@reduxjs/toolkit';

import {
  TypedUseSelectorHook,
  useDispatch as dispatchHook,
  useSelector as selectorHook
} from 'react-redux';

import appReducer from './slices/appSlice';
import ingredientReducer from './slices/ingredientSlice';
import orderReducer from './slices/orderSlice';
import userReducer from './slices/userSlice';
import feedReducers from './slices/feedSlice';
import userOrdersReducers from './slices/userOrdersSlice';

export const rootReducer = {
  app: appReducer,
  ingredients: ingredientReducer,
  order: orderReducer,
  user: userReducer,
  feed: feedReducers,
  userOrders: userOrdersReducers
};

const store = configureStore({
  reducer: rootReducer,
  devTools: process.env.NODE_ENV !== 'production'
});

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;

export const useDispatch: () => AppDispatch = () => dispatchHook();
export const useSelector: TypedUseSelectorHook<RootState> = selectorHook;

export default store;
