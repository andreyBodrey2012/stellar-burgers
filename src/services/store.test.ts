import { expect, describe, it } from '@jest/globals';
import { configureStore } from '@reduxjs/toolkit';
import { rootReducer } from './store';

describe('проверка rootReducer', () => {
  it('проверка должен возвращать корректное начальное состояние для неизвестного экшена', () => {
    const store = configureStore({
      reducer: rootReducer,
      devTools: false
    });
    store.dispatch({ type: 'UNKNOWN_ACTION' });

    expect(store.getState()).toMatchSnapshot();
  });
});
