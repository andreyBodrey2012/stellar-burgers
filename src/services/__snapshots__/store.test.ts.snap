// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка rootReducer проверка AppState 1`] = `
{
  "app": {
    "isLoadingOrder": false,
    "ready": false,
    "selectedFeedOrder": null,
    "selectedIngredient": null,
    "selectedOrder": null,
  },
  "feed": {
    "error": null,
    "isLoading": false,
    "items": [],
    "total": 0,
    "totalToday": 0,
  },
  "ingredients": {
    "error": null,
    "isLoading": false,
    "items": [],
  },
  "order": {
    "error": null,
    "items": {
      "bun": null,
      "ingredients": [],
    },
    "orderModalData": null,
    "orderRequest": false,
  },
  "user": {
    "error": null,
    "isInit": false,
    "isLoading": false,
    "user": null,
  },
  "userOrders": {
    "error": null,
    "isLoading": false,
    "items": [],
  },
}
`;

exports[`проверка rootReducer проверка должен возвращать корректное начальное состояние для неизвестного экшена 1`] = `
{
  "app": {
    "isLoadingOrder": false,
    "ready": false,
    "selectedFeedOrder": null,
    "selectedIngredient": null,
    "selectedOrder": null,
  },
  "feed": {
    "error": null,
    "isLoading": false,
    "items": [],
    "total": 0,
    "totalToday": 0,
  },
  "ingredients": {
    "error": null,
    "isLoading": false,
    "items": [],
  },
  "order": {
    "error": null,
    "items": {
      "bun": null,
      "ingredients": [],
    },
    "orderModalData": null,
    "orderRequest": false,
  },
  "user": {
    "error": null,
    "isInit": false,
    "isLoading": false,
    "user": null,
  },
  "userOrders": {
    "error": null,
    "isLoading": false,
    "items": [],
  },
}
`;
